auth/src/main/java/org/aerogear/mobile/security/SecurityCheckExecutor.java

```diff
@@ -27,30 +28,30 @@ private SecurityCheckExecutor() {}
         /**
          * Base class for SecurityCheckExecutor builders.
          *
-         * @param <T> The type of this builder.
-         * @param <K> The type of the built object.
+         * @param <T> The type of this builder
+         * @param <K> The type of the built object
          */
         static abstract class AbstractBuilder<T, K> {
             /**
-             * {@link Context} of the device.
+             * {@link Context} of the device. Cannot be null.
              */
             private final Context ctx;
 
             /**
-             * the {@link Collection<SecurityCheck>} of security checks to be tested.
+             * the {@link Collection<SecurityCheck>} of security checks to be tested. Can be null.
              */
             private final Collection<SecurityCheck> checks = new HashSet<>();
 
             /**
-             * {@link MetricsService}.
+             * {@link MetricsService}. Can be null.
              */
             private MetricsService metricsService;
 
             /**
              * Creates AbstractBuilder object.
              *
-             * @param ctx {@link Context} of the device.
-             * @throws IllegalArgumentException if ctx is null.
+             * @param ctx {@link Context} of the device
```

Comment:
Here we could specify that the parameter can be null, but, TBH I would specify only when the parameter CAN be null, because the opposite is what is generally expected.